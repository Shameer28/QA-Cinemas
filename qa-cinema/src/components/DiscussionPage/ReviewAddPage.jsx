
import { Button } from "bootstrap";
import {useState, useEffect} from "react"
import movieUtils from "../../utils/movies";

const ReviewAddPage = () => {

	const [stage, setStage] = useState(0);

	const [films, setFilms] = useState([]);

	const [film, setFilm] = useState({ title: "Detective Pikachu", _id: "123123123" });

	const [rating, setRating] = useState(-1);

	const [msg, setMsg] = useState("");

	const [name, setName] = useState("");

	const [email, setEmail] = useState("");


	useEffect( ()=> {
		movieUtils.getAll().then( (resp) => {
			setFilms(resp.data);
		})
	}, []);


	const selectFilm = (e, film) => {
		e.preventDefault();

		setFilm(film);
		setStage(1);
	}


	const msgUpdate = (e, str)=> {
		e.preventDefault();

		if (str.length > 300) {
			setMsg(msg);
			return;
		}

		setMsg(str);
	};


	const submit = (e)=> {
		e.preventDefault();

		// check if the film is valid

		if (rating < 1) {
			// enter a valid raiting
			return
		}
		if (msg.length == 0) {
			// add a valid review
			return
		}




		const reviewData = {
			rating: rating,
			name: name,
			email: email,
			msg: msg,
			filmId: film._id,
		}

		movieUtils.createRating(reviewData).then( () => {
			setStage(3);
		}).catch( (err) => {
			// Put this in an alert of something
			setStage(1);
		});
		setStage(2);
	};

	if (stage === 0) {
		return ( <div>
			<h4> Select a film to add a review</h4>

			<div>
				{ films.map(  (resp) => (
					<button onClick = {  (e) => {  selectFilm(e, resp) } }>
						{resp.title}
					</button>
				)) }
			</div>


		</div>
		);

	}else if (stage == 1) {


		const ratings = []

		for (let i = 1; i<6; i++) {
			ratings.push(<button onClick={ (event)=> {
				event.preventDefault();
				setRating(i);
			}
			 }>{  (rating === i) ? "Selected" : i}</button>)
		}

		return(
			<div>

				<form action="">
					<fieldset>
						<legend>Details</legend>
						<input value={name} type="text" placeholder="Name" onChange={ (e) => { e.preventDefault(); setName(e.target.value); } } />
						<input value={email} type="email" placeholder="Email Address" required onChange={ (e) => { e.preventDefault(); setEmail(e.target.value) } } />
					</fieldset>

					<h4>
						How many ratings do you want to give {film.title}?
					</h4>

					{ratings}

					<br/>
				
					<textarea value={msg} name="" id="" cols="30" rows="10" onChange={  (e)=> { msgUpdate(e, e.target.value )} }  ></textarea>
					
					<p>
						{msg.length} / 300
					</p>
					<br/>
					<button onClick= {submit}> Add Review</button>
				</form>
			</div>
		);
	}
	else if (stage == 2) {
		return ( <div>
			<p>
				Waiting ...
			</p>
		</div>)
	}
	else if (stage == 3) {
		return ( <div>
			<p>
				Rating added succesfully
			</p>
		</div>)
	}


	return ( <div>
		<p>
			Looks like something went wrong, Please refresh the page
		</p>	
	</div>);
}
 
export default ReviewAddPage;