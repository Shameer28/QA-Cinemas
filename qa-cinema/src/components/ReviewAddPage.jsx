
import {useState} from "react"

const ReviewAddPage = () => {

	const [stage, setStage] = useState(0);

	const [films, setFilms] = useState([]);

	if (stage === 0) {
		return ( <div>
			<h4> Select a film to add a review</h4>

			<div>
				<p>films</p>
				<p>films</p>
				<p>films</p>
				<p>films</p>
				<p>films</p>
				<p>films</p>
				<p>films</p>
				<p>films</p>
				<p>films</p>
			</div>


		</div>
		)

	}



	return ( <div>
		<p>
			ahasdasdasd
		</p>	
	</div>);
}
 
export default ReviewAddPage;