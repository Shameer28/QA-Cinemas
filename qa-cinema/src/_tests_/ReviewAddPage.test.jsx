import ReviewAddPage from "../components/DiscussionPage/ReviewAddPage";
import { create } from 'react-test-renderer';



describe(`Review Add Page testing`, () => {

    it(`Should match the snapshot`, () => {
        const instance = create(<ReviewAddPage />).toJSON();
        expect(instance).toMatchSnapshot();
    })
})