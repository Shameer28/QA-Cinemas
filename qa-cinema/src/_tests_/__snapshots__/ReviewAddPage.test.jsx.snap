// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Add Page testing Should match the snapshot 1`] = `
<div
  className="background"
>
  <img
    alt="Select Banner"
    className="img-fluid"
    src="review2.png"
    width="100%"
  />
  <div
    className="container"
  >
    <br />
    <h1
      id="revh1"
      style={
        Object {
          "color": "white",
          "fontWeight": "400",
          "fontfamily": "isonormregular, sans-serif",
          "letterSpacing": ".15em",
          "lineHeight": "1.1",
          "textTransform": "uppercase",
        }
      }
    >
       Select a film to add a review
    </h1>
    <div
      id="revcarddiv"
      style={
        Object {
          "height": "100%",
          "marginLeft": "auto",
          "marginRight": "auto",
          "width": "100%",
        }
      }
    >
      <br />
      <div
        className="row row-cols-md-4"
      />
    </div>
  </div>
</div>
`;
